src/engine.ml

```diff
@@ -1364,9 +1366,42 @@ let wrap_hmac_control now ts mtu session tls_auth key transport outs =
   in
   (session, transport, List.rev outs)
 
-let wrap_tls_crypt_control now ts mtu session tls_crypt key transport outs =
+let wrap_tls_crypt_control now ts mtu session (tls_crypt, wkc) needs_wkc key
+    transport outs =
   let now_ts = ptime_to_ts_exn now in
   let acks = bytes_of_acks transport in
+  (* If we reply with hmac cookie we must split such that the first control
+     packet, the Control_wkc has room for the /cleartext/ wkc, and fix the
+     packet length afterwards *)
+  let session, transport, maybe_out, outs =
+    match (needs_wkc, outs) with
+    | true, (`Control, data) :: rest ->
+        let l = min (mtu - Cstruct.length wkc) (Cstruct.length data) in
+        let data, data' = Cstruct.split data l in
+        let rest =
+          if Cstruct.is_empty data' then rest else (`Control, data') :: rest
+        in
```

Comment:
```suggestion
        (* To ensure we send back ACKs we unconditionally add another control *)
        let rest = (`Control, data') :: rest in
```
I need to check that an empty control packet doesn't cause issues, but maybe this is one way to fix the issue mentioned.